File: cpp/541_反转字符串_II_easy.cpp
class Solution {
public:
    string reverseStr(string s, int k) {
        int N = s.length();
        for(int i=0; i<N; i+=2*k){
            int j = i;
            int t = min(N-1, j+k-1);
            while(j<t){
                char temp = s[j];
                s[j++] = s[t];
                s[t--] = temp;
            }
        } 
        return s;
    }
    
};

File: python/541_反转字符串_II.easy.py
class Solution:
    def reverseStr(self, s: str, k: int) -> str:
        def reverse(s):
            # s = list(s)
            N = len(s)
            for i in range(0, (N+1)//2):
                s[i], s[N-i-1] = s[N-i-1], s[i]
            return s
        s = list(s)
        N = len(s)
        for i in range(0, N, 2*k):
            s[i:i+k] = reverse(s[i:i+k])
        return ''.join(s)